btservice/AndroidBluetoothServer.cpp

```diff
@@ -1,53 +1,214 @@
+AndroidBluetoothServer::AndroidBluetoothServer(openauto::configuration::IConfiguration::Pointer config_)
     : rfcommServer_(std::make_unique<QBluetoothServer>(QBluetoothServiceInfo::RfcommProtocol, this))
+    , config(std::move(config_))
 {
+    handshakeState = IDLE;
     connect(rfcommServer_.get(), &QBluetoothServer::newConnection, this, &AndroidBluetoothServer::onClientConnected);
+    QThread *thread = QThread::create([&]{ this->stateMachine(); });
+    thread->start();
+}
+
+void AndroidBluetoothServer::stateMachine()
+{
+    while(true){
+        switch(handshakeState){
+            case IDLE:
+                break;
+            
+            case DEVICE_CONNECTED:
+                handshakeState = SENDING_SOCKETINFO_MESSAGE;
+                break;
+
+            case SENDING_SOCKETINFO_MESSAGE:
+                writeSocketInfoMessage();
+                break;
+
+            case SENT_SOCKETINFO_MESSAGE:
+                break;
+
+            case PHONE_RESP_SOCKETINFO:
+                handshakeState = SENDING_NETWORKINFO_MESSAGE;
+                break;
+            
+            case SENDING_NETWORKINFO_MESSAGE:
+                writeNetworkInfoMessage();
+                break;
+            
+            case SENT_NETWORKINFO_MESSAGE:
+                break;
+
+            case PHONE_RESP_NETWORKINFO:
+                break;
+            
+        }
+    }
 }
 
 bool AndroidBluetoothServer::start(const QBluetoothAddress& address, uint16_t portNumber)
 {
+    OPENAUTO_LOG(info)<<"listening";
     return rfcommServer_->listen(address, portNumber);
 }
 
 void AndroidBluetoothServer::onClientConnected()
 {
-    auto socket = rfcommServer_->nextPendingConnection();
+    socket = rfcommServer_->nextPendingConnection();
 
     if(socket != nullptr)
     {
-        OPENAUTO_LOG(info) << "[AndroidBluetoothServer] rfcomm client connected, peer name: " << socket->peerName().toStdString();
+        OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Device Connected: " << socket->peerName().toStdString();
+        connect(socket, SIGNAL(readyRead()), this, SLOT(readSocket()));
+        handshakeState = DEVICE_CONNECTED;
     }
     else
     {
         OPENAUTO_LOG(error) << "[AndroidBluetoothServer] received null socket during client connection.";
     }
 }
 
+bool AndroidBluetoothServer::writeProtoMessage(uint16_t messageType, google::protobuf::Message &message){
+    QByteArray byteArray(message.SerializeAsString().c_str(), message.ByteSize());
+    uint16_t message_length = message.ByteSize();
+    char byte1 =  messageType & 0x000000ff;
+    char byte2 = (messageType & 0x0000ff00) >> 8;
+    byteArray.prepend(byte1);
+    byteArray.prepend(byte2);
+    byte1 =  message_length & 0x000000ff;
+    byte2 = (message_length & 0x0000ff00) >> 8;
+    byteArray.prepend(byte1);
+    byteArray.prepend(byte2);
+    int sentBytes = socket->write(byteArray);
+    if(sentBytes != byteArray.length()) return false;
```

Comment:
per the openauto styling, this conditional should be wrapped in brackets (on new lines)

Reply:
same goes for any of the brace-wrapped things (functions, switch-case, etc)